src/test/java/org/folio/rest/support/matchers/ResponseMatcher.java

```diff
@@ -0,0 +1,58 @@
+package org.folio.rest.support.matchers;
+
+import java.util.Objects;
+
+import org.folio.rest.support.Response;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeMatcher;
+
+import io.vertx.core.json.DecodeException;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+public class ResponseMatcher {
+
+  public static Matcher<Response> hasValidationError(
+    String expectedMessage, String expectedKey, String expectedValue) {
+
+    return new TypeSafeMatcher<Response>() {
+      @Override
+      protected boolean matchesSafely(Response response) {
+        if (response.getStatusCode() != 422) {
+          return false;
+        }
+
+        try {
+          JsonArray errors = response.getJson().getJsonArray("errors");
+          if (errors != null && errors.size() == 1) {
```

Comment:
Can it be mapped to Error object to avoid getting properties by name?

Reply:
Fixed.